### ISSUE TITLE
Replay File MD5 verification

### ISSUE THREAD

**Issue by Eve-ning** [ppy/osu-web#7954]
Summary: [ReplayFile.php::headerChunk()](https://github.com/ppy/osu-web/blob/28bb42f994996bcf31c97e6d6ebb195a6a94c034/app/Libraries/ReplayFile.php#L62) Replicating the following `md5` hashing for replays does not generate the same `md5` as in the replay

```php
$md5 = md5("{$score->maxcombo}osu{$user->username}{$beatmap->checksum}{$score->score}{$score->rank}");
```

## Python Replica

This is the python replica i tried to use

```python

import hashlib
import numpy as np
from osrparse import parse_replay_file

r = parse_replay_file("path/to/replay")

rmd = r.replay_hash
bmd = r.beatmap_hash
encodable = f"{r.max_combo}osu{r.player_name}{bmd}{r.score}S".encode('utf-8')
rmd_c = hashlib.md5(encodable).hexdigest()
```
requires: `pip install osrparse`

Debug Variables:
```python
bmd = {str} '946f34e226ce8589a0f9e861042e99d7'
encodable = {bytes: 53} b'2576osuEvening946f34e226ce8589a0f9e861042e99d7979004S'
r = {Replay} <osrparse.replay.Replay object at 0x0000016700769A88>
rmd = {str} '8b544ae13c649142116af97c1260613c'
rmd_c = {str} '34c20973b290eaa2fb3e7b570b6e66f7'
```
`rmd` and `rmd_c` does not match

### Request

Is it possible to debug output the non-md5-encoded `$md5` in [ReplayFile.php::headerChunk()](https://github.com/ppy/osu-web/blob/28bb42f994996bcf31c97e6d6ebb195a6a94c034/app/Libraries/ReplayFile.php#L62) through **any** test in [ReplayFileTest.php](https://github.com/ppy/osu-web/blob/28bb42f994996bcf31c97e6d6ebb195a6a94c034/tests/Libraries/ReplayFileTest.php) so I can verify my string format?

Thanks!

**Comment by peppy** [comment]
You can use `php artisan tinker` to run commands for testing.

**Comment by Eve-ning** [comment]
i have absolutely no idea how to use php unfortunately, i did follow the setup guide but i keep getting stuck, i think i would be fine if i can get the string to verify the format I'm using

**Comment by peppy** [comment]
Not sure sorry. This is out of scope of this project.
